core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -601,231 +457,330 @@ class SpendReportingServiceSpec
-
-      val e = intercept[RawlsExceptionWithErrorReport] {
-        Await.result(
-          service.getSpendForBillingProject(testData.billingProject.projectName,
-                                            startDate =
-                                              DateTime.now().minusDays(spendReportingServiceConfig.maxDateRange + 1),
-                                            endDate = DateTime.now()
-          ),
-          Duration.Inf
-        )
-      }
-      e.errorReport.statusCode shouldBe Option(StatusCodes.BadRequest)
-  }
+  it should "throw an exception if the billing project cannot be found" in {
+    val samDAO = mock[SamDAO]
+    when(samDAO.userHasAction(any(), any(), any(), any())).thenReturn(Future.successful(true))
+    val dataSource = mock[SlickDataSource]
+    when(dataSource.inTransaction[Option[BillingProjectSpendExport]](any(), any())).thenReturn(Future.successful(None))
+    val service = new SpendReportingService(
+      testContext,
+      dataSource,
+      Resource.pure[IO, GoogleBigQueryService[IO]](mock[GoogleBigQueryService[IO]]),
+      samDAO,
+      spendReportingServiceConfig
+    )
 
-  it should "throw an exception if the billing project cannot be found" in withDefaultTestDatabase {
-    dataSource: SlickDataSource =>
-      val service = createSpendReportingService(dataSource)
+    val e = intercept[RawlsExceptionWithErrorReport] {
+      Await.result(service.getSpendExportConfiguration(RawlsBillingProjectName("fakeProject")), Duration.Inf)
+    }
 
-      val e = intercept[RawlsExceptionWithErrorReport] {
-        Await.result(service.getSpendForBillingProject(RawlsBillingProjectName("fakeProject"),
-                                                       DateTime.now().minusDays(1),
-                                                       DateTime.now()
-                     ),
-                     Duration.Inf
-        )
-      }
-      e.errorReport.statusCode shouldBe Option(StatusCodes.NotFound)
+    e.errorReport.statusCode shouldBe Option(StatusCodes.NotFound)
+    e.errorReport.message shouldBe s"billing project fakeProject not found"
   }
 
-  it should "throw an exception if the billing project does not have a linked billing account" in withDefaultTestDatabase {
-    dataSource: SlickDataSource =>
-      val service = createSpendReportingService(dataSource)
-      val projectName = RawlsBillingProjectName("fakeProject")
-      runAndWait(
-        dataSource.dataAccess.rawlsBillingProjectQuery.create(
-          RawlsBillingProject(projectName, CreationStatuses.Ready, billingAccount = None, None)
-        )
-      )
+  it should "throw an exception if the billing project does not have a linked billing account" in {
+    val samDAO = mock[SamDAO]
+    when(samDAO.userHasAction(any(), any(), any(), any())).thenReturn(Future.successful(true))
+    val dataSource = mock[SlickDataSource]
+    when(dataSource.inTransaction[Option[BillingProjectSpendExport]](any(), any()))
```

Comment:
I'm unclear how this maps to the billing project not having a linked billing account.

Reply:
`getSpendExportConfiguration` is calling `RawlsBillingProject.getBillingProjectSpendConfiguration`, which calls

```scala
  def toBillingProjectSpendExport(projectRecord: RawlsBillingProjectRecord): BillingProjectSpendExport = {
    val table = for {
      googleProjectId <- projectRecord.spendReportDatasetGoogleProject
      datasetName <- projectRecord.spendReportDataset
      tableName <- projectRecord.spendReportTable
    } yield s"$googleProjectId.$datasetName.$tableName"

    val billingAccount = RawlsBillingAccountName(
      projectRecord.billingAccount.getOrElse(
        throw new RawlsException(s"billing account not set on project ${projectRecord.projectName}")
      )
    )

    BillingProjectSpendExport(RawlsBillingProjectName(projectRecord.projectName), billingAccount, table)
  }
```

So if the billing account isn't set, it throws the exception, which `getSpendExportConfiguration` is looking for.
It isn't completely clear to me why this is divided up where it is,  but from a behavior perspective, this does mock how the underlying call behaves.

I'm not completely certain how `RawlsExceptionWithErrorReport`s are handled when the ErrorReport doesn't have a status code set (I would assume it would default to 400). If that is the case, we could just not catch the exception in `getSpendExportConfiguration`, and let it bubble up, which would give the same behavior, and let us eliminate this test entirely. That actually sounds like a good idea....

Reply:
I _think_ we default to 500 judging by the default exception handler code [here](https://github.com/broadinstitute/rawls/blob/develop/core/src/main/scala/org/broadinstitute/dsde/rawls/webservice/RawlsApiService.scala#L48)